src/components/DisplayArticles.js

```diff
@@ -2,46 +2,106 @@ import React, { Component } from "react";
 import axios from "axios";
 import ArticleTemplate from "./ArticleTemplate";
 import { Container } from "semantic-ui-react";
+import { connect } from 'react-redux'
+import getCity from "../modules/openCageWrapper";
+import getLocalArticles from "../modules/getLocalArticles"
 
 
 class DisplayArticles extends Component {
   state = {
-      articles: []
-    };
-
+    articles: [],
+    position: {},   
+    city: "",
+    localArticles: []
+  };
 
   componentDidMount() {
     this.getArticles();
+
+    navigator.geolocation.getCurrentPosition(position => {
+      this.setState({ position: position }, () => {
+        this.getUserCity();
+      });
+    });
+  }
+
+  async getUserCity() {
+    let city = await getCity(
+      this.state.position.coords.latitude,
+      this.state.position.coords.longitude
+    );
+    this.setState({ city: city }, () => {
+      if (this.state.city) {
+        this.getLocalArticles();
+      }
+    }); 
+  }
+
+  async getLocalArticles(){
+    let response = await getLocalArticles(this.state.city);
+
+    if (response.status === 200) {
+      this.setState({
+        localArticles: response.data
+      });
+    } else {
+      return response.error
```

Comment:
where do we return this error?
maybe flash a message with the error

Reply:
fixed